chain/db/pubsub.go

```diff
@@ -238,14 +277,14 @@ func (sdb *StateDB) getSubscribers(topic string, bucket uint32) (map[string]stri
 
 		subscriberString := address.MakeAddressString(ps.subscriber, ps.identifier)
 
-		subscribers[subscriberString] = ps.meta
+		subscribers = append(subscribers, subscriberString)
```

Comment:
We are fundamentally changing the API here. IMO it's better to have an API or an option that can get both subscribers and metadata like before. These info are already read into the memory here, so it will be much more efficient if we can just include it in the response. If I'm using subscribers to store service providers' info, the price and other info will be in the meta. If I want to make a choice, I definitely need to know the price of every provider. With the current new API, I have to call getSubscription() for each provider, which is not feasible at all if there are hundreds of providers.

Reply:
Problem is that returning meta for all subscribers could be A LOT of data, since we don't limit max page size and meta limit is quite big too. It could be an option with no meta by default.

BTW for tuna I'm choosing random provider and go with him, if price is ok for me. I don't search for the lowest, because that way providers can control who can connect to them (to attack them somehow, for example). So, I prefer random here with max acceptable price set.

Reply:
We can make it by default not returning meta. Another thing we can have is a configurable parameter specifying max response size for (probably any) rpc, and if response is larger than it, just return error.

Also for tuna, random provider can work as the first version, but later I think it will be better to combine provider IP + price to choose the one that is (approximately) closest and cheapest.